eigen_pod_proofs.go

```diff
@@ -63,32 +63,18 @@ func (epp *EigenPodProofs) ComputeBeaconStateRoot(beaconState *deneb.BeaconState
 	return beaconStateRoot, nil
 }
 
-func (epp *EigenPodProofs) ComputeBeaconStateTopLevelRoots(beaconState *deneb.BeaconState) (*beacon.BeaconStateTopLevelRoots, error) {
-	beaconStateTopLevelRootsSlice, err := epp.loadOrComputeBeaconData(
-		BEACON_STATE_TOP_LEVEL_ROOTS_PREFIX,
-		beaconState.Slot,
-		func() ([]byte, error) {
-			beaconStateTopLevelRoots, err := beacon.ComputeBeaconStateTopLevelRootsDeneb(beaconState)
-			if err != nil {
-				return nil, err
-			}
-			return json.Marshal(beaconStateTopLevelRoots)
-		},
-	)
+func (epp *EigenPodProofs) ComputeBeaconStateTopLevelRoots(beaconState *spec.VersionedBeaconState) (*beacon.BeaconStateTopLevelRoots, error) {
+	//get the versioned beacon state's slot
+	slot, err := beaconState.Slot()
 	if err != nil {
 		return nil, err
 	}
-	beaconStateTopLevelRoots := &beacon.BeaconStateTopLevelRoots{}
-	err = json.Unmarshal(beaconStateTopLevelRootsSlice, beaconStateTopLevelRoots)
-	return beaconStateTopLevelRoots, err
-}
 
-func (epp *EigenPodProofs) ComputeBeaconStateTopLevelRootsCapella(beaconState *capella.BeaconState) (*beacon.BeaconStateTopLevelRoots, error) {
 	beaconStateTopLevelRootsSlice, err := epp.loadOrComputeBeaconData(
 		BEACON_STATE_TOP_LEVEL_ROOTS_PREFIX,
-		beaconState.Slot,
+		slot,
 		func() ([]byte, error) {
-			beaconStateTopLevelRoots, err := beacon.ComputeBeaconStateTopLevelRootsCapella(beaconState)
+			beaconStateTopLevelRoots, err := epp.ComputeVersionedBeaconStateTopLevelRoots(beaconState)
```

Comment:
`ComputeVersionedBeaconStateTopLevelRoots` accepts a `spec.VersionedBeaconState` as an arg, so you may rename this function to `ComputeBeaconStateTopLevelRoots` since the arg already identifies the type used to compute such value

Reply:
I think this makes it more explicit, stating that its versioned is important I think